src/ucp/proto/proto_am.inl

```diff
@@ -158,10 +160,12 @@ void ucp_dt_iov_copy_uct(ucp_context_h context, uct_iov_t *iov, size_t *iovcnt,
         dst_it                      = 0;
         state->dt.iov.iov_offset    = 0;
         while ((dst_it < max_dst_iov) && (src_it < max_src_iov)) {
-            if (src_iov[src_it].length) {
+            if (src_iov[src_it].length != 0) {
                 iov[dst_it].buffer  = UCS_PTR_BYTE_OFFSET(src_iov[src_it].buffer, iov_offset);
                 iov[dst_it].length  = src_iov[src_it].length - iov_offset;
-                iov[dst_it].memh    = state->dt.iov.dt_reg[src_it].memh[0];
+                iov[dst_it].memh    = ((md_flags & UCT_MD_FLAG_NEED_MEMH) ?
```

Comment:
does it help to avoid registration somewhere?

Reply:
This fix prevents crash that occurs in UCP AM tests with TCP.

Reply:
BTW, maybe better allocate dt_reg in either way, but memh by `UCT_MEM_HANDLE_NULL` to avoid extra branch here
@brminich wdyt?